File: _includes/code/llms-txt/python/crud.py
"""llms.txt snippet: object CRUD. Section "Python / TypeScript > CRUD"."""
import os
import weaviate
from weaviate.classes.init import Auth
from weaviate.classes.config import Configure

client = weaviate.connect_to_weaviate_cloud(
    cluster_url=os.environ["WEAVIATE_URL"],
    auth_credentials=Auth.api_key(os.environ["WEAVIATE_API_KEY"]),
)
client.collections.delete("Movie__CrudPy")
client.collections.create(
    "Movie__CrudPy",
    vector_config=Configure.Vectors.text2vec_weaviate(),
)

# START llms_crud
movies = client.collections.use("Movie__CrudPy")

# Create — insert one object, returns its UUID
uuid = movies.data.insert({"title": "Inception", "genre": "Science Fiction"})

# Read — fetch the object by its UUID
obj = movies.query.fetch_object_by_id(uuid)
print(obj.properties)

# Update — merge new property values into the object
movies.data.update(uuid=uuid, properties={"genre": "Sci-Fi Thriller"})

# Delete — remove the object by its UUID
movies.data.delete_by_id(uuid)
# END llms_crud

assert movies.query.fetch_object_by_id(uuid) is None
client.collections.delete("Movie__CrudPy")
client.close()

File: _includes/code/llms-txt/typescript/crud.ts
// llms.txt snippet: object CRUD. Section "Python / TypeScript > CRUD".
import weaviate, { vectors } from 'weaviate-client';

const client = await weaviate.connectToWeaviateCloud(process.env.WEAVIATE_URL!, {
  authCredentials: new weaviate.ApiKey(process.env.WEAVIATE_API_KEY!),
});
await client.collections.delete('Movie__CrudTs');
await client.collections.create({
  name: 'Movie__CrudTs',
  vectorizers: vectors.text2VecWeaviate(),
});

// START llms_crud
const movies = client.collections.use('Movie__CrudTs');

// Create — insert one object, returns its UUID
const uuid = await movies.data.insert({ title: 'Inception', genre: 'Science Fiction' });

// Read — fetch the object by its UUID
const obj = await movies.query.fetchObjectById(uuid);
console.log(obj?.properties);

// Update — merge new property values into the object
await movies.data.update({ id: uuid, properties: { genre: 'Sci-Fi Thriller' } });

// Delete — remove the object by its UUID
await movies.data.deleteById(uuid);
// END llms_crud

if (await movies.query.fetchObjectById(uuid)) throw new Error('object not deleted');
await client.collections.delete('Movie__CrudTs');
await client.close();
